### PR TITLE
Add retry logic for app review submission

### PR DESCRIPTION
**Pull request by nid90**
See: https://github.com/fastlane/fastlane/issues/20065#issuecomment-1086977589

Handles: https://tramline.sentry.io/issues/4460775239/?project=4504591889268736&query=is%3Aunresolved&referrer=issue-stream&stream_index=0

### CODE REVIEW

File: lib/app_store/connect.rb [hunk 1]
```diff
@@ -224,8 +228,27 @@ def create_review_submission(build_number:, version: nil)
         submission ||= app.create_review_submission(platform: IOS_PLATFORM)
 
         submission.add_app_store_version_to_review_items(app_store_version_id: edit_version.id)
+
+        do_submit(submission, edit_version)
+      end
+    end
+
+    def do_submit(submission, edit_version)
```
Comment (kitallis) [inline lib/app_store/connect.rb:236]:
```suggestion
    def submit_review(submission, edit_version)
```

File: lib/app_store/connect.rb [hunk 2]
```diff
@@ -205,7 +207,9 @@ def create_review_submission(build_number:, version: nil)
       execute do
         build = get_build(build_number)
 
-        edit_version = app.get_edit_app_store_version(includes: "build")
+        edit_version = app
+          .get_app_store_versions(includes: "build", filter: INFLIGHT_RELEASE_FILTERS)
+          .find { |v| v.build&.version == build_number.to_s }
```
Comment (kitallis) [inline lib/app_store/connect.rb:212]: I feel like this is repeated a lot, can we extract?

**Review by kitallis — APPROVED** [review]
looks mostly good

File: lib/app_store/connect.rb (hunk 2 above)
Comment (nid90) [reply lib/app_store/connect.rb:213]: this is actually not repeated.

**Review by kitallis — CHANGES_REQUESTED** [review]

**Review by kitallis — APPROVED** [review]

**Pull request merged**